txscript/opcode.go

```diff
@@ -1850,7 +1850,17 @@ func opcodeCheckSig(op *parsedOpcode, vm *Engine) error {
 		return nil
 	}
 
-	ok := signature.Verify(hash, pubKey)
+	var ok bool
+	if vm.sigCache != nil {
+		ok = vm.sigCache.Exists(hash, signature, pubKey)
```

Comment:
I've opted to uniformly use `valid` both here and in `opcodeCheckMultiSig`. But, I don't see any gains to be hand by pushing out the fragment to a new function.